baybe/utils/plotting.py

```diff
@@ -0,0 +1,56 @@
+"""Plotting utilities."""
+
+import json
+import os
+import warnings
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+
+
+def create_plots(data: pd.DataFrame, path: Path, base_name: str, **kwargs) -> None:
+    """Create plots from a given data frame and save them as a svg file.
+
+    The plots will be saved in the location specified by ``path``.
+    The attribute ``base_name`` is used to define the name of the outputs.
+
+    If the ``SMOKE_TEST`` variable is set, no plots are being saved.
+
+    Using the ``BAYBE_MULTIVERSION_PLOTS`` environment variable, it is possible to
+    create plots for the light and dark version of the documentation. If this variable
+    is not set, a single file named ``{base_name}_check.svg`` is created.
+
+    Args:
+        data: The data frame containing the data to be plotted.
+        path: The path of the file that should be saved.
+        base_name: The base name that is used for naming the output files.
+        **kwargs: Keyword arguments. They are directly passed to ``sns.lineplot`` and
+            are used for specifying the plot.
+    """
+    # File containing all the themes
+    themes = json.load(open(Path("plotting_themes.json")))
+    # Environment variables for checking whether we want to have multiversion plots
+    BAYBE_MULTIVERSION_PLOTS = "BAYBE_MULTIVERSION_PLOTS" in os.environ
+
+    # Choose either all available or just the `check` theme.
+    chosen_themes = themes if BAYBE_MULTIVERSION_PLOTS else ["check"]
+    for theme_name in chosen_themes:
+        # Extract and set the current theme
+        theme = themes[theme_name]
+        font_scale, rc_params = theme["font_scale"], theme["rc_params"]
+        sns.set_theme(style="ticks", font_scale=font_scale, rc=rc_params)
+        # Only if kwargs are being provided, a plot is actually created
+        if kwargs:
+            sns.lineplot(data=data, **kwargs)
+            output_path = Path(path, f"{base_name}_{theme_name}.svg")
+            if "SMOKE_TEST" not in os.environ:
```

Comment:
I suggest to undo your last commit, because the way SMOKE_TEST was handled before was actually the better approach. It's called a "guard clause", it has the benefit of 1) reducing indentation level 2) exiting fast 3) better readability of the code logic. From your git message I understand that you did it due to coverage, but these are really two things that need to be separated completely.

Reply:
Ok, but then we need to exclude this file from being checked by coverage as we always hace SMOKE_TEST=true in these.

Reply:
it seems like a reasonable exclude as this code contains stuff that should explicitly be skipped in testing

Reply:
Then I'll change back and exclude :)

Reply:
ONe question though: What is the best way to exclude a single file from the coverage? It seems like there is no configuration file for this and that the tests just happen in our CI pipeline, do you know how we can easily change this?

Reply:
see https://github.com/emdgroup/baybe/pull/79/files#diff-834e9b406d74791ffbafaeec9cc894082cea9739bc347b6ff9f72312c92ebc79